tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -1,77 +1,83 @@
-﻿// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
-
-using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Globalization;
-using System.IO;
-using System.Runtime.InteropServices;
-using System.Text;
-using System.Text.Json.Serialization;
-using System.Text.RegularExpressions;
-using System.Threading.Tasks;
-using Microsoft.UI.Xaml;
-using Microsoft.UI.Xaml.Controls;
-using Serilog;
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using DevHome.Common.Extensions;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Serilog;
 using Windows.System;
 using Windows.Win32;
 using Windows.Win32.Foundation;
-using Windows.Win32.System.Com;
+using Windows.Win32.System.Com;
 using Windows.Win32.UI.WindowsAndMessaging;
-using static DevHome.PI.Helpers.WindowHelper;
-
-namespace DevHome.PI.Helpers;
-
-public enum ToolActivationType
-{
+using static DevHome.PI.Helpers.WindowHelper;
+
+namespace DevHome.PI.Helpers;
+
+public enum ToolActivationType
+{
     Protocol,
     Msix,
-    Launch,
-}
-
-public partial class ExternalTool : Tool
+    Launch,
+}
+
+public partial class ExternalTool : Tool
 {
     private static readonly ILogger _log = Log.ForContext("SourceContext", nameof(ExternalTool));
     private readonly string _errorTitleText = CommonHelper.GetLocalizedString("ToolLaunchErrorTitle");
 
     private readonly string _errorMessageText = CommonHelper.GetLocalizedString("ToolLaunchErrorMessage");
 
-    public string ID { get; private set; }
-
+    public string ID { get; private set; }
+
     public string Executable { get; private set; }
 
-    [JsonConverter(typeof(EnumStringConverter<ToolActivationType>))]
-    public ToolActivationType ActivationType { get; private set; } = ToolActivationType.Launch;
-
-    public string Arguments { get; private set; }
+    [JsonConverter(typeof(EnumStringConverter<ToolActivationType>))]
+    public ToolActivationType ActivationType { get; private set; } = ToolActivationType.Launch;
+
+    public string Arguments { get; private set; }
+
+    public string ExtraInfo { get; private set; }
 
     public string AppUserModelId { get; private set; }
 
     public string IconFilePath { get; private set; }
 
-    public ExternalTool(
-        string name,
-        string executable,
+    public ExternalTool(
+        string name,

```

Comment:
I understand this code isn't new. But C# is not Javascript and we should steer away from Javascript code patterns like giving (mostly) all the parameters defaults.  I understand this is not a trivial change because this behavior is already in PI and all users of ExternalTool understand some values might be empty.  Just something to watch out for.

Reply:
Yeah. I think in this particular case it was used because this is a JSON serialized object, and this goofiness is part of getting the deserializer to work.